sljit_src/sljitNativeMIPS_32.c

```diff
@@ -26,6 +26,25 @@
 
 /* mips 32-bit arch dependent functions. */
 
+static sljit_s32 emit_copysign(struct sljit_compiler *compiler, sljit_s32 op,
+		sljit_sw src1, sljit_sw src2, sljit_sw dst)
+{
+	int is_32 = (op & SLJIT_32);
+	sljit_ins fr_pair = is_32 ? 0 : (1 << 11);
+
+	if (!is_32 && src1 != dst)
+		FAIL_IF(push_inst(compiler, MFC1 | T(TMP_REG1) | FS(src1), DR(TMP_REG1)));
```

Comment:
One more thing. You can directly copy single precision registers, you don't need to move them into a cpu register. It is also one less instruction.

Reply:
Put this copy before the XOR to improve parallel execution in the cpu pipeline.